File: Localisator_Demo-objC/Localisator_Demo-objC/ViewController.m
//
//  ViewController.m
//  Localisator_Demo-objC
//
//  Created by Michaël Azevedo on 10/02/2015.
//  Copyright (c) 2015 Michaël Azevedo. All rights reserved.
//

#import "ViewController.h"
#import "Localisator.h"
#import "LocalisatorViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UILabel *homeTitleLabel;
@property (weak, nonatomic) IBOutlet UILabel *homeDescLabel;
@property (weak, nonatomic) IBOutlet UIButton *languageButton;

@end

@implementation ViewController

#pragma mark - Init methods

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.title = @"Localisator";
    
    [[NSNotificationCenter defaultCenter] addObserver:self
                                             selector:@selector(receiveLanguageChangedNotification:)
                                                 name:kNotificationLanguageChanged
                                               object:nil];
    
    [self configureViewFromLocalisation];
}

-(void)configureViewFromLocalisation
{
    [self.homeTitleLabel    setText:LOCALIZATION(@"HomeTitleText")];
    [self.homeDescLabel     setText:LOCALIZATION(@"HomeDescText")];
    
    [self.languageButton setTitle:LOCALIZATION(@"HomeButtonTitle") forState:UIControlStateNormal];
}

#pragma mark - Notification methods

- (void) receiveLanguageChangedNotification:(NSNotification *) notification
{
    if ([notification.name isEqualToString:kNotificationLanguageChanged])
    {
        [self configureViewFromLocalisation];
    }
}

#pragma mark - Memory management

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(void)dealloc
{
    [[NSNotificationCenter defaultCenter] removeObserver:self name:kNotificationLanguageChanged object:nil];
}


@end

File: Localisator_Demo-swift/Localisator_Demo-swift/ViewController.swift
//
//  ViewController.swift
//  Localisator_Demo-swift
//
//  Created by Michaël Azevedo on 10/02/2015.
//  Copyright (c) 2015 Michaël Azevedo. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    
    // MARK: - Outlets
    
    @IBOutlet var homeTitleLabel: UILabel!
    @IBOutlet var homeDescLabel: UILabel!
    @IBOutlet var languageButton: UIButton!
    
     // MARK: - Init methods
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        title = "Localisator"
        
        NotificationCenter.default.addObserver(self, selector: #selector(ViewController.receiveLanguageChangedNotification(notification:)), name: kNotificationLanguageChanged, object: nil)
        
        configureViewFromLocalisation()
    }
    
    func configureViewFromLocalisation() {
    
        homeTitleLabel.text     = Localization("HomeTitleText")
        homeDescLabel.text      = Localization("HomeDescText")
        languageButton.setTitle(Localization("HomeButtonTitle"), for: UIControlState.normal)
        
    }

    // MARK: - Notification methods
    
    func receiveLanguageChangedNotification(notification:NSNotification) {
        if notification.name == kNotificationLanguageChanged {
            configureViewFromLocalisation()
        }
    }
    
    // MARK: - Memory management
    
    deinit {
        NotificationCenter.default.removeObserver(self, name: kNotificationLanguageChanged, object: nil)
    }

}

